RecoTracker/MkFitCMS/src/MkStdSeqs.cc

```diff
@@ -0,0 +1,586 @@
+        // int  n_ovlp_tracks = 0;
+
+        //#pragma simd /* Vectorization via simd had issues with icc */
+        for (int tss = ts + 1; tss < ns; tss++) {
+          const float Pt2 = pt[tss];
+
+          ////// Always require charge consistency. If different charge is assigned, do not remove seed-track
+          if (charge[tss] != charge[ts])
+            continue;
+
+          const float thisDPt = std::abs(Pt2 - Pt1);
+          ////// Require pT consistency between seeds. If dpT is large, do not remove seed-track.
+          if (thisDPt > dpt_common * (Pt1))
+            continue;
+
+          const float Eta2 = eta[tss];
+          const float deta2 = std::pow(Eta1 - Eta2, 2);
+
+          const float oldPhi2 = oldPhi[tss];
+
+          const float pos2_second = pos2[tss];
+          const float thisDXYSign05 = pos2_second > pos2_first ? -0.5f : 0.5f;
+
+          const float thisDXY = thisDXYSign05 * sqrt(std::pow(x[ts] - x[tss], 2) + std::pow(y[ts] - y[tss], 2));
+
+          const float invptq_second = invptq[tss];
+
+          const float newPhi1 = oldPhi1 - thisDXY * invR1GeV * invptq_first;
+          const float newPhi2 = oldPhi2 + thisDXY * invR1GeV * invptq_second;
+
+          const float dphi = cdist(std::abs(newPhi1 - newPhi2));
+
+          const float dr2 = deta2 + dphi * dphi;
+
+          const float thisDZ = z[ts] - z[tss] - thisDXY * (ctheta[ts] + ctheta[tss]);
+          const float dz2 = thisDZ * thisDZ;
+
+          ////// Reject tracks within dR-dz elliptical window.
+          ////// Adaptive thresholds, based on observation that duplicates are more abundant at large pseudo-rapidity and low track pT
+          bool overlapping = false;
+          if (std::abs(Eta1) < etamax_brl) {
+            if (Pt1 > ptmin_hpt) {
+              if (dz2 * dzmax2_inv_bh + dr2 * drmax2_inv_bh < 1.0f)
+                overlapping = true;
+            } else {
+              if (dz2 * dzmax2_inv_bl + dr2 * drmax2_inv_bl < 1.0f)
+                overlapping = true;
+            }
+          } else {
+            if (Pt1 > ptmin_hpt) {
+              if (dz2 * dzmax2_inv_eh + dr2 * drmax2_inv_eh < 1.0f)
+                overlapping = true;
+            } else {
+              if (dz2 * dzmax2_inv_el + dr2 * drmax2_inv_el < 1.0f)
+                overlapping = true;
+            }
+          }
+
+          if (overlapping) {
+            //Mark tss as a duplicate
+            i1 = ts;
+            i2 = tss;
+            if (d0[tss] > d0[ts])
+              writetrack[tss] = false;
+            else {
+              writetrack[ts] = false;
+              i2 = ts;
+              i1 = tss;
+            }
+            // Add hits from tk2 to the seed we are keeping.
+            // NOTE: We only have 3 bits in Track::Status for number of seed hits.
+            //       There is a check at entry and after adding of a new hit.
+            Track &tk = seeds[i1];
+            if (merge_hits && tk.nTotalHits() < 15) {
```

Comment:
could this magic number 15 (also seems repeated below) be made a constexpr somewhere?

Reply:
Addressed in Round 1 follow-up PR trackreco/cmssw#75 commit: https://github.com/trackreco/cmssw/pull/75/commits/3859c79ead0d8a60ee14a8299107ab767ed54bd8